app/models/spree/gateway/uqpay_china_union.rb

```diff
@@ -36,6 +36,21 @@ def method_type
       "uqpay_china_union"
     end
 
+    def cancel(payment_source)
+      response = self.refund({
```

Comment:
You can't just call refund here, you need to check the status first.
Refund is only possible if the status is Success,
If the status is Paying You'll need to call cancel (Check first I'm not quite sure)

Reply:
Status checking is already handled by Spree. Refund only processed on completed payments.